app/src/main/java/com/android/voyageur/ui/trip/activities/ActivitiesMapScreen.kt

```diff
@@ -0,0 +1,275 @@
+  var searchQuery by remember { mutableStateOf("") }
+  var selectedDate by remember { mutableStateOf<Date?>(null) }
+
+  /** Filters the activities based on the search query, selected filters, and selected date. */
+  fun applyFilters() {
+    activities =
+        tripsViewModel.getActivitiesForSelectedTrip().filter { activity ->
+          val matchesQuery =
+              activity.title.contains(searchQuery, ignoreCase = true) ||
+                  activity.description.contains(searchQuery, ignoreCase = true)
+          val matchesFilters =
+              selectedFilters.isEmpty() ||
+                  selectedFilters.contains(activity.activityType.toString())
+          val matchesDate =
+              selectedDate == null ||
+                  activity.startTime.toDate().let { date ->
+                    date.year == selectedDate!!.year &&
+                        date.month == selectedDate!!.month &&
+                        date.date == selectedDate!!.date
+                  }
+          matchesQuery && matchesFilters && matchesDate
+        }
+    onActivitiesChanged(activities)
+  }
+
+  Column {
+    // Search Bar with Filter Button
+    Row(
+        modifier = Modifier.fillMaxWidth().padding(16.dp),
+        verticalAlignment = Alignment.CenterVertically) {
+          SearchBar(
+              placeholderId = R.string.search_activities,
+              onQueryChange = { query ->
+                searchQuery = query
+                applyFilters()
+              },
+              modifier = Modifier.weight(1f).testTag("searchBar"))
+          Spacer(modifier = Modifier.width(8.dp))
+          IconButton(
+              modifier = Modifier.testTag("filterButton"), onClick = { showFilterDialog = true }) {
+                Icon(
+                    imageVector = Icons.Outlined.FilterAlt,
+                    contentDescription = stringResource(R.string.filter_activities),
+                    tint = MaterialTheme.colorScheme.primary)
+              }
+        }
+
+    // Google Map
+    Box(
+        modifier =
+            Modifier.fillMaxSize()
+                .padding(16.dp)
+                .clip(MaterialTheme.shapes.medium)
+                .background(MaterialTheme.colorScheme.secondary)) {
+          GoogleMap(
+              modifier = Modifier.fillMaxSize().testTag("GoogleMap"),
+              cameraPositionState = cameraPositionState) {
+                activities.forEach { activity ->
+                  Marker(
+                      state =
+                          MarkerState(
+                              position = LatLng(activity.location.lat, activity.location.lng)),
+                      title = activity.title,
+                      snippet = activity.description)
+                }
+              }
+        }
+  }
+
+  // Filter Dialog
+  if (showFilterDialog) {
+    FilterDialog(
+        onFilterByDay = { date ->
+          selectedDate = date
+          applyFilters()
+          showFilterDialog = false
```

Comment:
When filtering after type/date or clear the filters, I suggest that we don't close the primary dialog automatically, as if an user wants to filter by both, he would have to reopen the filters. Let me know your thoughts.

Reply:
Fixed